pynestml/codegeneration/resources_nest/point_neuron/common/NeuronClass.jinja2

```diff
@@ -537,45 +561,34 @@ void {{neuronName}}::handle(nest::DataLoggingRequest& e)
 void {{neuronName}}::handle(nest::SpikeEvent &e)
 {
   assert(e.get_delay_steps() > 0);
-{%- if neuron.is_multisynapse_spikes() %}
-{%-   set port = neuron.get_spike_input_ports()[0] %}
-  B_.{{port.get_symbol_name()}}[e.get_rport() - 1].add_value(
-    e.get_rel_delivery_steps( nest::kernel().simulation_manager.get_slice_origin() ),
-    e.get_weight() * e.get_multiplicity() );
-{%- elif neuron.get_multiple_receptors()|length > 1 %}
   assert( e.get_rport() < static_cast< int >( B_.spike_inputs_.size() ) );
 
-  B_.spike_inputs_[ e.get_rport() ].add_value(
-    e.get_rel_delivery_steps( nest::kernel().simulation_manager.get_slice_origin() ),
-    e.get_weight() * e.get_multiplicity() );
-{%- else %}
-  const double weight = e.get_weight();
-  const double multiplicity = e.get_multiplicity();
-{%-   for port in neuron.get_spike_input_ports() %}
-{%-     if port.is_excitatory() and port.is_inhibitory() %}
-  // this port receives both excitatory and inhibitory spikes (where inhibitory spikes are those where weight < 0)
-  get_{{port.get_symbol_name()}}().
-      add_value(e.get_rel_delivery_steps( nest::kernel().simulation_manager.get_slice_origin()),
-                     weight * multiplicity );
-{%-     elif port.is_excitatory() %}
-  // this port receives excitatory spikes
+  double weight = e.get_weight();
+  int nestml_port = 0;
+{%- if neuron.get_spike_input_ports()|length > 1 %}
   if ( weight >= 0.0 )
   {
-    get_{{port.get_symbol_name()}}().
-        add_value(e.get_rel_delivery_steps( nest::kernel().simulation_manager.get_slice_origin()),
-                       weight * multiplicity );
+    if( rport_to_nestml_port.find(std::make_pair(e.get_rport(), "spike")) != rport_to_nestml_port.end() )
```

Comment:
Why not make the mapping the other way around, so the call to ``find()`` is not necessary?

Reply:
Since we have to route spikes based on `rport` in the event via `e.get_rport()`, the map has the `rport` as the key. How do you suggest otherwise?

Reply:
OK, sorry, my previous suggestion was not very helpful, but I'm still worried about the runtime performance hit of using ``std::map.find()``, because this method will be called whenever an incoming spike is handled, so quite a lot during runtime. Possibly, we have to do some more low-level C-style array indexing magic.

For instance, ``rport_to_nestml_port`` could be a plain C array where the integer index is the NEST rport. There could be a second boolean array of the same length which says if this input port uses the same or separate entries for exc/inh spikes. In case of inhibitory spikes, we would just add 1 to the index. This is less "clean" coding-wise but may be necessary.

@jougs: do you have any suggestion here, should we benchmark the current implementation first? Or do you agree to reject the use of ``find()`` offhand?

Reply:
I have now modified the mapping of `rport` numbers with `std::vector` instead of a `std::map`.
@clinssen Could you please have a look?